code/modules/projectiles/guns/energy.dm

```diff
@@ -43,10 +43,13 @@
 
 /obj/item/gun/energy/Initialize(mapload)
 	. = ..()
-	if(cell_type)	//just in case someone eventually wants to make an energy gun without a power cell for some reason?
+	if(cell_type)
 		cell = new cell_type(src)
-		cell.maxcharge = gun_charge
-		cell.charge = gun_charge		//allows fine-tuned control over how much charge a gun will have without defining new types of cells
+		if(gun_charge) //But we only use this if it is defined instead of overwriting every cell to 1000 by default like a dumbass
+			cell.maxcharge = gun_charge
+			cell.charge = gun_charge
+	else
+		cell = new(src)
```

Comment:
```suggestion
		else
			cell = new(src)
```

Reply:
No, this is correct how it is

Old code, in case there was no cell:
```
if(cell_type)	//just in case someone eventually wants to make an energy gun without a power cell for some reason?
	cell = new cell_type(src)
else
		cell = new(src)
```

The else is a safety in case there is no cell defined. I took it out in my refactor and tbh I'm not sure why I did.
Putting it as an else after this would cause issues. 
```
cell = new cell_type(src)
if(gun_charge)
```

Reply:
Wouldn't this recursively cause energy guns to be created?
From how I read this, it will create a new energy gun as the cell, if there is no cell available.

Reply:
> Wouldn't this recursively cause guns to be created? From how I read this, it will create a new energy gun as the cell, if there is no cell available.